logstash-core/src/test/java/org/logstash/ClonerTest.java

```diff
@@ -35,12 +35,28 @@ public void testRubyStringCloning() {
         // Check object identity
         assertTrue(result != original);
 
-        // Check different underlying bytes
-        assertTrue(result.getByteList() != original.getByteList());
-
         // Check string equality
         assertEquals(result, original);
 
         assertEquals(javaString, result.asJavaString());
+
+        // NOTE: this is an implementation detail or the underlying sharing:
+        assertTrue(result.getByteList() == original.getByteList());
+    }
+
+    @Test
+    public void testRubyStringCloningAndAppend() {
```

Comment:
I think we should validate the behaviour in _both_ directions, mutating the original and the copy independently to make sure we don't affect the other.

Reply:
added a separate test